client/src/amplify-ui/code-block-switcher/code-block-switcher.tsx

```diff
@@ -0,0 +1,81 @@
+import {Component, Host, h, Prop, State} from "@stencil/core";
+import {
+  hostStyle,
+  activeTabStyle,
+  tabStyle,
+  tabContainerStyle,
+} from "./code-block-switcher.style";
+import {css} from "emotion";
+
+const getTabLocalStorageKey = (tabHeadingList: string): string =>
+  `amplify-docs::${tabHeadingList}`;
+
+@Component({tag: "amplify-code-block-switcher", shadow: false})
+export class AmplifyCodeBlockSwitcher {
+  /** the tab headings, comma-separated as a single string */
+  @Prop() readonly tabHeadingList?: string;
+
+  @State() activeChildI = 0;
+
+  tabsHeadings: string[];
+
+  tabLocalStorageKey: string;
+
+  componentWillLoad() {
+    if (this.tabHeadingList) {
+      // get the split array of headings, to map over (for rendering buttons)
+      this.tabsHeadings = this.tabHeadingList.split(",").map((e) => e.trim());
+      // set the local storage key, so we can persist tab selection cross-session
+      this.tabLocalStorageKey = getTabLocalStorageKey(this.tabHeadingList);
+      // get from the previous session, if present
+      const persistedActiveChildI =
+        localStorage.getItem(this.tabLocalStorageKey) || undefined;
+      // ... if persisted selection is present...
+      if (typeof persistedActiveChildI === "string") {
+        // set the active child in state
+        this.activeChildI = parseInt(persistedActiveChildI);
+      }
+    }
+  }
+
+  createActiveChildISetter = (i: number) => (_event: Event): void => {
+    this.activeChildI = i;
+    localStorage.setItem(this.tabLocalStorageKey, String(i));
+  };
+
+  render() {
+    // + 2 because CSS child indices start at 1, and the first child in the host is the div of tabs
+    const revealStyle = css`
+      > pre:nth-child(${this.activeChildI + 2}) {
```

Comment:
How does the "first child in the host is the div of tabs" in a selector for `pre` tags?

Reply:
Inspect the DOM. The elements passed in via slot are rendered adjacent to the tab-containing div. As such, the `nth-child` selector needs to be passed the tab index plus 1. We add another because `nth-child` starts with 1, whereas arrays with 0 (ofc). Didn't mean to devsplain there haha.

Reply:
Ok, we should use `pre:nth-of-type(${this.activeChildI})` and set the initial value to 1. Unless there is a good reason to not follow the CSS child indices order